### PR TITLE
Caching: Added new picture and tweaked text

### PR DESCRIPTION
**Pull request by roryhewitt**
#917

### CODE REVIEW

File: src/content/en/2020/caching.md [hunk 1]
```diff
@@ -15,11 +15,11 @@ discuss: 2056
...
 queries: 20_Caching
 #featured_quote: Caching benefits both the end users (they get their web pages quickly) and the companies serving the web pages (reducing the load on their servers). Caching really is a win-win!
-#featured_stat_1: TODO
+#featured_stat_1: **One quarter** of all HTTP responses don't include any caching information, which means they're probably being cached wrongly
 #featured_stat_label_1: TODO
-#featured_stat_2: TODO
+#featured_stat_2: More than **one fifth** of all responses aren't using revalidation, resulting in wasted bandwidth and increased latency
 #featured_stat_label_2: TODO
-#featured_stat_3: TODO
+#featured_stat_3: Our chapter has a cat picture!
```
Comment (rviscomi) [inline src/content/en/2020/caching.md:22]: 😸 Can we use a real stat here instead?

**Review by rviscomi — CHANGES_REQUESTED** [review]

File: src/content/en/2020/caching.md (hunk 1 above)
Comment (roryhewitt) [reply src/content/en/2020/caching.md:22]: We could. I was trying to add some humor :) I'll dig out a useful one...

Comment (roryhewitt) [reply src/content/en/2020/caching.md:22]: @rviscomi what should go in the `featured_stat_label` fields?

Comment (rviscomi) [reply src/content/en/2020/caching.md]: Here's an example of how it looks on the home page

image

The big text like 72.58% is the stat and "percent of length values..." is the label.

Comment (roryhewitt) [reply src/content/en/2020/caching.md]: Ah - ok then, I'll change it - currently I have the `stat` just the text string. More commits!

Comment (roryhewitt) [reply src/content/en/2020/caching.md]: Also, it looks like there's an error trying to compress at least one of the images - probably the new `luna-cat.png` image... Any thoughts on what I should do? I can make it smaller, if that helps - right now it's about 2MB.

File: src/content/en/2020/caching.md [hunk 2]
```diff
@@ -8,19 +8,19 @@ authors: [roryhewitt, raghuramakrishnan71]
 reviewers: [csswizardry, jzyang, jaisanth, Soham-S-Sarkar]
```
Comment (bazzadp) [inline src/content/en/2020/caching.md:8]: Did csswizardy contribute? I ping him earlier and he wasn't sure he had!

File: src/content/en/2020/caching.md [hunk 3]
```diff
@@ -8,19 +8,19 @@ authors: [roryhewitt, raghuramakrishnan71]
...
-#featured_stat_label_2: TODO
-#featured_stat_3: TODO
-#featured_stat_label_3: TODO
+#featured_quote: Caching benefits both end users (they get their web pages quickly) and the companies serving web pages (reducing the load on their servers). Caching really is a win-win!
+#featured_stat_1: 25.6
+#featured_stat_label_1: Number of HTTP responses don't include any caching information, which means they're probably being cached wrongly
+#featured_stat_2: 21.4%
+#featured_stat_label_2: Number of responses that aren't using revalidation, resulting in wasted bandwidth and increased latency
+#featured_stat_3: 78.6%
+#featured_stat_label_3: Number of sites could save up to 2MB of bandwidth by improving their caching policy
```
Comment (bazzadp) [inline src/content/en/2020/caching.md:23]: Can activate these now:

```suggestion
featured_quote: Caching benefits both end users (they get their web pages quickly) and the companies serving web pages (reducing the load on their servers). Caching really is a win-win!
featured_stat_1: 25.6%
featured_stat_label_1: Number of HTTP responses don't include any caching information, which means they're probably being cached wrongly
featured_stat_2: 21.4%
featured_stat_label_2: Number of responses that aren't using revalidation, resulting in wasted bandwidth and increased latency
featured_stat_3: 78.6%
featured_stat_label_3: Number of sites could save up to 2MB of bandwidth by improving their caching policy
```

Also this is what it looks like now:

Featured quote

The quotes itself is quite short compared to the labels.

File: src/content/en/2020/caching.md [hunk 4]
```diff
@@ -94,6 +94,14 @@ When considering what to cache, it is important to understand whether the respon
 * An example of static content is an image.  For instance, a picture of a cat is the same regardless of who's requesting it or where the requester is located.
 * An example of dynamic content is a list of events which are specific to a geographic location. The list will be different based on the requester's location.
 
+{{ figure_markup(
+  image="luna-cat.png",
+  link="https://almanac.httparchive.org/static/images/2020/caching/luna-cat.png",
```
Comment (bazzadp) [inline src/content/en/2020/caching.md:99]: The link happens automatically so no need to include this. It's meant for if you want a different link for some reason.

```suggestion
```

**Review by bazzadp — CHANGES_REQUESTED** [review]

**Comment by roryhewitt** [comment]
@bazzadp Ach, my laptop is throwing a fit. Please go ahead with your recommended changes. I don't *think* @csswizardry reviewed - in fact, the only person that I **know** reviewed was @zjyang.

 Is your suggestion that I should reduce the text length of each stat, or increase the length of the quote?

File: src/content/en/2020/caching.md [hunk 5]
```diff
@@ -94,6 +94,14 @@ When considering what to cache, it is important to understand whether the respon
 * An example of static content is an image.  For instance, a picture of a cat is the same regardless of who's requesting it or where the requester is located.
 * An example of dynamic content is a list of events which are specific to a geographic location. The list will be different based on the requester's location.
 
+{{ figure_markup(
+  image="luna-cat.png",
```
Comment (rviscomi) [inline src/content/en/2020/caching.md:98]: Is this really worth the 2.19 MB?

File: src/content/en/2020/caching.md (hunk 3 above)
Comment (rviscomi) [reply src/content/en/2020/caching.md:23]: I'd also suggest rewording "Number" to "Percent"

**Comment by roryhewitt** [comment]
@rviscomi - reduced size of cat image. Can reduce smaller, but would really love to keep it in there somehow. Also changed from "Number" to "Percentage" and increased the featured quote so it's not as small compared to the stats.


I would have responded directly above, but I can't use Chrome today and Edge won't allow me to enter anything in the other text boxes:(

File: src/content/en/2020/caching.md [hunk 6]
```diff
@@ -8,19 +8,19 @@ authors: [roryhewitt, raghuramakrishnan71]
...
-#featured_stat_label_2: TODO
-#featured_stat_3: TODO
-#featured_stat_label_3: TODO
+#featured_quote: Caching provides a significant performance benefit by avoiding costly network requests - it helps both end users (they get their web pages quickly) and the companies serving web pages (reducing the load on their servers). Caching really is a win-win!
+#featured_stat_1: 25.6
+#featured_stat_label_1: Percentage of HTTP responses don't include any caching information, which means they're probably being cached wrongly
+#featured_stat_2: 21.4%
+#featured_stat_label_2: Percentage of responses that aren't using revalidation, resulting in wasted bandwidth and increased latency
+#featured_stat_3: 78.6%
+#featured_stat_label_3: Percentage of sites could save up to 2MB of bandwidth by improving their caching policy
```
Comment (rviscomi) [inline src/content/en/2020/caching.md]: Reapplying @bazzadp's suggested changes
```suggestion
featured_quote: Caching provides a significant performance benefit by avoiding costly network requests - it helps both end users (they get their web pages quickly) and the companies serving web pages (reducing the load on their servers). Caching really is a win-win!
featured_stat_1: 25.6%
featured_stat_label_1: Percentage of HTTP responses don't include any caching information, which means they're probably being cached wrongly
featured_stat_2: 21.4%
featured_stat_label_2: Percentage of responses that aren't using revalidation, resulting in wasted bandwidth and increased latency
featured_stat_3: 78.6%
featured_stat_label_3: Percentage of sites could save up to 2MB of bandwidth by improving their caching policy
```

File: src/content/en/2020/caching.md [hunk 7]
```diff
@@ -8,19 +8,19 @@ authors: [roryhewitt, raghuramakrishnan71]
 reviewers: [csswizardry, jzyang, jaisanth, Soham-S-Sarkar]
 analysts: [raghuramakrishnan71]
 translators: []
-roryhewitt_bio: Enterprise Architect at <a href="https://www.akamai.com/">Akamai</a>. Passionate about performance. Competition-level snowboarder. Multiple black belts. Motorcyclist. Boxer. Troublemaker. 
+roryhewitt_bio: Enterprise Architect at <a href="https://www.akamai.com/">Akamai</a>, who is passionate about performance. A British ex-patriate, he has lived in San Francisco for more than twenty years. In his spare time, he's a long-distance adventure motorcyclist, competition-level snowboarder and boxer with multiple black belts in various styles of karate. He likes being known as a troublemaker. Most importantly, he's a father and husband and the owner of Luna the cat. Find him at <a href="https://www.linkedin.com/in/roryhewitt/">LinkedIn</a>.
```
Comment (rviscomi) [inline src/content/en/2020/caching.md:11]: FYI you could also add your LinkedIn profile to your [contributor config](https://github.com/HTTPArchive/almanac.httparchive.org/blob/main/src/config/2020.json#L1131)

File: src/content/en/2020/caching.md (hunk 2 above)
Comment (rviscomi) [reply src/content/en/2020/caching.md:8]: IIUC @jzyang was the only reviewer? If so let's update to this.
```suggestion
reviewers: [jzyang]
```

**Comment by roryhewitt** [comment]
@rviscomi I *think* @jzyang was the only reviewer. None of the others contacted me to let me know they were actively reviewing it. Should I ask them individually, or just assume that if I didn't hear from them, they didn't do anything?

**Comment by bazzadp** [comment]
> @rviscomi - reduced size of cat image. Can reduce smaller, but would really love to keep it in there somehow. Also changed from "Number" to "Percentage" and increased the featured quote so it's not as small compared to the stats.
> 
> I would have responded directly above, but I can't use Chrome today and Edge won't allow me to enter anything in the other text boxes:(

Tinypng.com does the following:

image

Happy to add to this branch if you want?

**Comment by rviscomi** [comment]
@roryhewitt yes, you would have noticed other reviewers suggesting edits or being more involved so I agree with the change to only keep @jzyang.

**Comment by roryhewitt** [comment]
OK, cat image is small, @jzyang is the only reviewer, and I think all the suggested changes have been made. Does the 'ratio' of featured quote size to stat labels look OK to you?

File: src/content/en/2020/caching.md [hunk 8]
```diff
@@ -1,26 +1,26 @@
...
-#featured_stat_label_2: TODO
-#featured_stat_3: TODO
-#featured_stat_label_3: TODO
+featured_quote: Caching provides a significant performance benefit by avoiding costly network requests - it helps both end users (they get their web pages quickly) and the companies serving web pages (reducing the load on their servers). Caching really is a win-win!
+featured_stat_1: 25.6%
+featured_stat_label_1: Percentage of HTTP responses don't include any caching information, which means they're probably being cached wrongly
+featured_stat_2: 21.4%
+featured_stat_label_2: Percentage of responses that aren't using revalidation, resulting in wasted bandwidth and increased latency
+featured_stat_3: 78.6%
+featured_stat_label_3: Percentage of sites could save up to 2MB of bandwidth by improving their caching policy
```
Comment (bazzadp) [inline src/content/en/2020/caching.md:23]: This is what it looks like:

image

I think a bit of a mouthful to be honest! Difficult to digest those stats quickly. Also some repetition in there (do we need to say it's a percentage when they have a percentage sign?)

Finally I think the 3rd stat is the wrong way of looking at this! Up to 2Mb of saving could mean 0.0001 Mb of saving. Let's flip it around and look at the ones over 2Mb instead of under (I also think we should change the chapter text to say this too).

```suggestion
featured_quote: Caching provides a significant performance benefit by avoiding costly network requests - it helps both end users (they get their web pages quickly) and the companies serving web pages (reducing the load on their servers). Caching really is a win-win!
featured_stat_1: 25.6%
featured_stat_label_1: HTTP responses with no caching information
featured_stat_2: 21.4%
featured_stat_label_2: Responses that cannot be revalidated
featured_stat_3: 21.3%
featured_stat_label_3: Sites that could save over 2MB on repeat visits with better caching
```

This looks like this, which I think is better:

image

So I also think you should change this line in the text:

> Lighthouse also indicates how many bytes could be saved on repeat views by enabling a longer cache policy. Of the sites that could benefit from additional caching, 78.6% of them can reduce their page weight by up to 2MB!

to this:

> Lighthouse also indicates how many bytes could be saved on repeat views by enabling a longer cache policy. Of the sites that could benefit from additional caching, 21.3% of them can reduce their page weight by over 2MB!

**Comment by roryhewitt** [comment]
@bazzadp  added your suggested edits. Greatly appreciated - I really need a firm editor :)

**Comment by bazzadp** [comment]
> @bazzadp added your suggested edits. Greatly appreciated - I really need a firm editor :)

Editing will come as soon as we get some space to breath - you're on our list - see #1432

**Review by bazzadp — APPROVED** [review]
Add the compressed images (compressed them all as some looked big but not much saving other than the cat one).

I'm happy to merge now if you are @rviscomi ?

**Comment by bazzadp** [comment]
Oh and also removed the unused contributors from the 2020.json file.

**Review by rviscomi — APPROVED** [review]
